src/Adapter/HttpClientPhpArrayFixture.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace Kununu\DataFixtures\Adapter;
+
+use Kununu\DataFixtures\Exception\InvalidFileException;
+use Kununu\DataFixtures\Tools\FixturesHttpClientInterface;
+use SplFileInfo;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Throwable;
+
+abstract class HttpClientPhpArrayFixture implements HttpClientFixtureInterface
+{
+    private $triedToLoadFeatures = false;
+
+    final public function load(HttpClientInterface $httpClient): void
+    {
+        if (!is_a($httpClient, FixturesHttpClientInterface::class)) {
+            $this->triedToLoadFeatures = false;
+
+            return;
+        }
+
+        foreach ($this->fileNames() as $fileName) {
+            $file = new SplFileInfo($fileName);
+
+            if ($file->getExtension() !== 'php') {
+                continue;
+            }
+
+            $httpClient->addResponses($this->loadFile($fileName));
+        }
+
+        $this->triedToLoadFeatures = true;
+    }
+
+    final public function triedToLoadFeatures(): bool
```

Comment:
From what I am seeing the intent of having this public method is because of the tests. Right?
If yes, then I suggest to remove it since the interface HttpClientFixtureInterface does not exposes and its not relevant to the consumers of the package / fixture, as it is an internal thing. 

Ideas:

- Add a method getResponses() to the FixturesHttpClientInterface and in the test use a stub instead of mock of the HttpClient
- In case there is no other way to test what you want without accessing it, then I would rather take advantage of _reflection_ (in particular https://www.php.net/manual/en/class.reflectionproperty.php) in the tests.

Reply:
Yes, because of the test :cry: